### PR TITLE
Update to replace AT_ERROR with TORCH_CHECK

### PR DESCRIPTION
**Pull request by lware1m1s**
Fixes #{52699}

### CODE REVIEW

File: c10/core/Backend.h [hunk 1]
```diff
@@ -142,7 +142,7 @@ static inline Backend dispatchKeyToBackend(DispatchKey t) {
   } else if (t == DispatchKey::Undefined) {
     return Backend::Undefined;
   } else {
-    AT_ERROR("Unrecognized tensor type ID: ", t);
+    TORCH_CHECK(false,"Unrecognized tensor type ID: ", t);
```
Comment (malfet) [inline c10/core/Backend.h:145]: Please add space between function/macro arguments. I.e. `foo(1, 2)` rather `foo(1,2)`
```suggestion
    TORCH_CHECK(false, "Unrecognized tensor type ID: ", t);
```

**Review by malfet — CHANGES_REQUESTED** [review]
Mostly looks good, but please add space after `false,`

File: c10/core/Backend.h (hunk 1 above)
Comment (lware1m1s) [reply c10/core/Backend.h]: @malfet fixed spacing issue lastest commit.

File: c10/cuda/CUDACachingAllocator.cpp [hunk 2]
```diff
@@ -614,7 +614,7 @@ class DeviceCachingAllocator {
     } else if (block->pool == &large_blocks) {
       return remaining > kSmallSize;
     } else {
-      AT_ERROR("should_split: invalid pool");
+      TORCH_CHECK(false,"should_split: invalid pool");
```
Comment (malfet) [inline c10/cuda/CUDACachingAllocator.cpp:617]: nit
```suggestion
      TORCH_CHECK(false, "should_split: invalid pool");
```

File: c10/cuda/CUDACachingAllocator.cpp [hunk 3]
```diff
@@ -921,7 +921,7 @@ class THCCachingAllocator {
   {
     Block* block = get_allocated_block(ptr);
     if (!block) {
-      AT_ERROR("invalid device pointer: ", ptr);
+      TORCH_CHECK(false,"invalid device pointer: ", ptr);
```
Comment (malfet) [inline c10/cuda/CUDACachingAllocator.cpp:924]:
```suggestion
      TORCH_CHECK(false, "invalid device pointer: ", ptr);
```

File: c10/cuda/CUDACachingAllocator.cpp [hunk 4]
```diff
@@ -887,7 +887,7 @@ class THCCachingAllocator {
     }
     Block* block = get_allocated_block(ptr, true /* remove */);
     if (!block) {
-      AT_ERROR("invalid device pointer: ", ptr);
+      TORCH_CHECK(false,"invalid device pointer: ", ptr);
```
Comment (malfet) [inline c10/cuda/CUDACachingAllocator.cpp:890]:
```suggestion
      TORCH_CHECK(false, "invalid device pointer: ", ptr);
```

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**